pinot-core/src/main/java/org/apache/pinot/core/geospatial/transform/function/ScalarFunctions.java

```diff
@@ -37,6 +37,13 @@ public class ScalarFunctions {
   private ScalarFunctions() {
   }
 
+  /**
+   * Maximum number of cells that can be returned by the gridDisk function. This is configurable via the system property
+   * "pinot.geospatial.gridDisk.maxCells". Default value is 100,000 cells.
+   */
+  public static final int MAX_GRID_DISK_CELLS = Integer.parseInt(
```

Comment:
Pinot doesn't usually read config from system properties. If you really want to bound this, consider making it a broker config, and reject the query when `k` is too large. You can find examples in `BaseSingleStageBrokerRequestHandler`. One drawback is that it is not honored for multi-stage engine